### PR TITLE
Nginx integration up/down migration

### PR DESCRIPTION
**Pull request by wjordan**
This PR enables a clean transition for when the `nginx_enabled` value is changed on an existing server.
- Add extra listeners/notifiers to ensure that the app services (`dashboard`/`pegasus`) are fully restarted with updated configuration whenever the `[app]_sock` and/or `[app]_port` configuration is changed. (A full restart is necessary because Unicorn will hang onto existing listeners through a USR2 upgrade signal.) These listeners make use of the new `:before` [notification timer](https://docs.chef.io/resource_common.html#timers) recently added to Chef-client in v12.6.0.
- Add a `cdo-nginx::stop` recipe to stop the existing nginx service (useful if `nginx_enabled` is reverted from `true` back to `false`).

### CODE REVIEW

File: cookbooks/cdo-nginx/metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license          'All rights reserved'
 description      'Installs/Configures cdo-nginx'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version          '0.0.3'
+version          '0.0.5'
```
Comment (wjordan) [inline cookbooks/cdo-nginx/metadata.rb:7]: I was using v `0.0.4` of this cookbook for development-testing purposes, so i thought I'd just bump this one version above that.

**Comment by islemaster** [comment]
LGTM, as far as I understand it :grin:.

File: cookbooks/cdo-secrets/recipes/globals.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
-template "/home/#{node[:current_user]}/#{node.chef_environment}/globals.yml" do
+template 'globals' do
```
Comment (islemaster) [inline cookbooks/cdo-secrets/recipes/globals.rb:1]: Did you just have to name this so you can refer to it from cdo_apps.rb#96 above?

Comment (wjordan) [inline cookbooks/cdo-secrets/recipes/globals.rb:1]: yup, exactly- I shortened the name here just so I didn't need to duplicate the whole path when referencing the resource above.

**Comment by wjordan** [comment]
Thanks for taking a look @islemaster! Now that I've spent more time with our chef scripts I'll try and put together an intro/refresher course for our team at some point- for now an extra pair of eyes on this diff is all I was after~

**Pull request merged**